charmcraft/config.py

```diff
@@ -164,6 +185,29 @@ class CharmhubConfig(
     storage_url: pydantic.HttpUrl = "https://storage.snapcraftcontent.com"
 
 
+class Base(
+    pydantic.BaseModel, extra=pydantic.Extra.forbid, frozen=True, validate_all=True
```

Comment:
I wonder if repeating this again and again, we could have something like...

````
class ConfigBaseModel(pydantic.BaseModel, extra=pydantic.Extra.forbid, frozen=True, validate_all=True):
    pass
```

And then just use that to inherit all our classes.